untitled1/src/com/aca/homework/week4/time/Time.java

```diff
@@ -0,0 +1,75 @@
+package com.aca.homework.week4.time;
+/*Create a Time method that takes seconds of a day in a constructor (starting from 00:00:00) and provides 3 method
+ to get hours, minutes and seconds of the time.
+        Time time = new Time(0)
+        time.getSeconds() -> should return 0
+        time.getMinutes() -> should return 0
+        time.getHours() -> should return 0
+
+
+        Time time = new Time(3600+5*60+6)
+        time.getSeconds() -> should return 6
+        time.getMinutes() -> should return 5
+        time.getHours() -> should return 1
+
+        print the result of each method invocation
+*/
+public class Time {
+    private long seconds;
+
+    public Time() {
+        seconds = 0;
+    }
+
+    public Time(long seconds) {
+        this.seconds = seconds - (seconds / (24 * 60 * 60)) * (24 * 60 * 60);
+    }
+
+    public long getSeconds() {
+        return seconds - getHours() * 3600 - getMinutes() * 60;
+    }
+
+    public long getMinutes() {
+        return (seconds - getHours() * 3600) / 60;
+    }
+
+    public long getHours() {
+        return seconds / 3600;
+    }
+
+    public void print() {
```

Comment:
the method is too big and is not readable

Reply:
I realized that it is not a good version but since we have different outputs for 8 cases, I did not find an optimal solution.
 public void print1() {
        long hours = getHours();
        long minutes = getMinutes();
        long seconds = getSeconds();
        boolean checkHours = checkHours(hours);
        boolean checkMinutes = checkMinutes(minutes);
        boolean checkSeconds = checkSeconds(seconds);

        if (checkHours && checkMinutes && checkSeconds)
            System.out.println("0" + hours + ":0" + minutes + ":0" + seconds);
        if (checkHours && checkMinutes && !checkSeconds)
            System.out.println("0" + hours + ":0" + minutes + ":" + seconds);
        if (checkHours && !checkMinutes && checkSeconds)
            System.out.println("0" + hours + ":" + minutes + ":0" + seconds);
        if (checkHours && !checkMinutes && !checkSeconds)
            System.out.println("0" + hours + ":" + minutes + ":" + seconds);
        if (!checkHours && checkMinutes && checkSeconds)
            System.out.println(hours + ":0" + minutes + ":0" + seconds);
        if (!checkHours && checkMinutes && !checkSeconds)
            System.out.println(hours + ":0" + minutes + ":" + seconds);
        if (!checkHours && !checkMinutes && checkSeconds)
            System.out.println(hours + ":" + minutes + ":0" + seconds);
        else
            System.out.println(hours + ":" + minutes + ":" + seconds);
    }

    public static boolean checkHours(long hours) {
        return hours >= 0 && hours <= 9;
    }

    public static boolean checkMinutes(long minutes) {
        return minutes >= 0 && minutes <= 9;
    }

    public static boolean checkSeconds(long seconds) {
        return seconds >= 0 && seconds <= 9;
    }
 
In this case, the code may be more readable.

Reply:
you can have a method that takes integer hours and returns a string.

return hours<=9 && hours>=0 ? "0"+hours : hours;

same for others and your code will be much smaller

Reply:
I wrote but it does not work,regardless of the argument range, 0 does not add from the front after calling pattern functions․The problem is with the pattern functions ․I checked it in main()

Reply:
This is a really good solution ․ Thank you very much